### PR TITLE
Add support for other aws partitions by tokenizing regions

### PR DESCRIPTION
**Pull request by fdamstra**
Based off @jordiup 's work on adding partition support:
* Tokenizes the region and determines partition based on that. I think this will be more future proof than a static list of regions.
* Fixes the console URL, which differs by partition.

Tested and appears functional. Gets me a valid console in both govcloud and commercial. I am unable to test china or the ISOs.

Supercedes #146 to solve #138

### CODE REVIEW

File: pkg/browsers/console.go [hunk 1]
```diff
@@ -281,8 +357,7 @@ func makeDestinationURL(service string, region string) (string, error) {
 		// and this avoids the need to keep the ServiceMap alphabetically sorted when developing Granted.
 		sort.Strings(validServices)
 
-		return "", fmt.Errorf("\nservice %s not found, please enter a valid service shortcut.\nValid service shortcuts: [%s]\n", service, strings.Join(validServices, ", "))
-
+		return "", fmt.Errorf("\nservice %s not found, please enter a valid service shortcut.\nValid service shortcuts: [%s]", service, strings.Join(validServices, ", "))
```
Comment (chrnorm) [inline pkg/browsers/console.go:360]: 👍 on this. Given that we're return an `err` here we don't need to worry about a newline at the end of the statement (I think the original implementation was a typo)

File: pkg/browsers/console.go [hunk 2]
```diff
@@ -184,15 +184,90 @@ const (
...
+	switch p {
+	case Default:
+		return "https://console.aws.amazon.com/"
+	case Gov:
+		return "https://console.amazonaws-us-gov.com/"
+	case Cn:
+		return "https://console.amazonaws.cn/"
+	}
+	// Note: we're not handling the ISO and ISOB cases, I don't think they are supported by a public AWS console
+	return "console.aws.amazon.com"
```
Comment (chrnorm) [inline pkg/browsers/console.go:236]: Might be a typo here.

```suggestion
	return "https://console.aws.amazon.com/"
```

File: pkg/browsers/console.go [hunk 3]
```diff
@@ -184,15 +184,90 @@ const (
...
+}
+
+func GetPartitionFromRegion(region string) PartitionHost {
```
Comment (chrnorm) [inline pkg/browsers/console.go:239]: Looks good. I am a fan of this approach!

**Comment by chrnorm** [comment]
🎉 congratulations on the first contribution @fdamstra! This looks good - thanks a lot for helping Granted support other partitions. I've taken a quick look over the changes - @jordiup could you please have a look too?

File: pkg/browsers/console.go (hunk 1 above)
Comment (fdamstra) [reply pkg/browsers/console.go:360]: This was @jordiup 's work from his original PR.

File: pkg/browsers/console.go (hunk 2 above)
Comment (fdamstra) [reply pkg/browsers/console.go:236]: Good catch. Will update the PR.

File: pkg/browsers/console.go [hunk 4]
```diff
@@ -184,15 +184,90 @@ const (
...
+
 	sessJSON, err := json.Marshal(sess)
+
 	if err != nil {
 		return "", err
 	}
 
+	partition := GetPartitionFromRegion(region)
+	// couldn't figure out how to flag this based on the '--verbose' setting
+	//fmt.Fprintf(color.Error, "Partition is detected as %s for region %s...\n", partition, region)
```
[the remark this answers is not in the document]
Comment (fdamstra) [reply pkg/browsers/console.go]: I've updated the code, adding the required first field of debug.Verbosity, but it doesn't output with the `--verbose` flag. Still not clear how this is supposed to work.

**Comment by jordiup** [comment]
Thanks for the help @fdamstra, I'm really happy with the changes, you smashed it. I'll merge now ✅

**Pull request merged**